### DISCUSSION TITLE
What do you use for linting?

### DISCUSSION THREAD

**Discussion by marclaporte** [smarty-php/smarty#951] [General]
The Tiki community uses https://github.com/umakantp/SmartyLint but the last commit is in 2015 and we have to patch it.

- https://gitlab.com/tikiwiki/tiki/-/blob/master/installer/composer-patches/smartylint-php8.1-and-other-fixes.patch
- https://github.com/umakantp/SmartyLint/issues/3

I saw this: https://github.com/smarty-php/smarty/issues/270

Thoughts?

**Comment by marclaporte** [comment]
Related: https://github.com/smarty-php/smarty/discussions/1136

**Comment by 3n9** [comment]
I built [this](https://github.com/3n9/SmartyLint) for myself.  It's mostly rubbish AI but it effectively meets my basic linting needs on the command line.

**Reply by gnutix** [reply]
Looks nice! Any chance you might add a MIT/Apache-2.0 license (Claude is adamant about this), so we could freely use it / contribute to it ? I submitted a few _AI-rubbish_ :wink:  PRs for some missing features that crash on some projects I have access to.

**Reply by 3n9** [reply]
> Looks nice! Any chance you might add a MIT/Apache-2.0 license (Claude is adamant about this), so we could freely use it / contribute to it ? I submitted a few _AI-rubbish_ 😉 PRs for some missing features that crash on some projects I have access to.

Done.

**Comment by marclaporte** [comment]
Great news!

https://github.com/umakantp/SmartyLint is now packaged on https://packagist.org/packages/smarty/smarty-lint
